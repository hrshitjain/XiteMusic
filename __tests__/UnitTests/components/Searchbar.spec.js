import React from 'react';
import { render } from '@testing-library/react-native';
import SearchBar from '../../../app/Components/Searchbar';

const defaultProps = {
    setSearchPhrase: jest.fn(),
}

describe('Searchbar component render', () =>{

    it('Snapshot test', () =>{
        const {toJSON} = render(<SearchBar {...defaultProps}/>);
        expect(toJSON()).toMatchSnapshot();
    });
    it('Root View has 1 child', () => {
        const {toJSON} = render(<SearchBar {...defaultProps}/>);
        expect(toJSON().children.length).toBe(1);
    });
    it('Component has 2 children', () => {
        const {toJSON} = render(<SearchBar {...defaultProps}/>);
        expect(toJSON().children[0].children.length).toBe(2);
    });
    it('Text Input is rendered',()=>{
        const {getAllByPlaceholderText} = render(<SearchBar {...defaultProps}/>);
        expect(getAllByPlaceholderText('Artist, music ...').length).toBe(1);
    });
});