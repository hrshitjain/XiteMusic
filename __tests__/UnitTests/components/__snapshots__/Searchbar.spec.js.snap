// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar component render Snapshot test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "margin": 10,
      "width": "80%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#d9dbda",
        "borderRadius": 15,
        "flexDirection": "row",
        "padding": 10,
        "width": "95%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 20,
          },
          Object {
            "marginLeft": 1,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï”…
    </Text>
    <TextInput
      clearButtonMode="while-editing"
      onChangeText={[MockFunction]}
      placeholder="Artist, music ..."
      style={
        Object {
          "fontSize": 20,
          "marginLeft": 10,
          "width": "90%",
        }
      }
    />
  </View>
</View>
`;
